cfy_manager/main.py

```diff
@@ -543,45 +547,54 @@ def _print_finish_message(config_file=None):
 def _are_components_installed():
     return all(
         os.path.isfile(os.path.join(INITIAL_INSTALL_DIR, service_name))
-        for service_name in get_installed_services_names())
+        for service_name in config[SERVICES_TO_INSTALL])
 
 
 def _are_components_configured():
     return all(
         os.path.isfile(os.path.join(INITIAL_CONFIGURE_DIR, service_name))
-        for service_name in get_installed_services_names())
+        for service_name in config[SERVICES_TO_INSTALL])
 
 
 def is_supervisord_service():
     return service._get_service_type() == 'supervisord'
 
 
-def _create_initial_install_file():
+def _create_initial_install_file(installed_components_list):
     """
-    Create /etc/cloudify/.installed/<service_name> if the
-    service installation finished successfully for the first time
+    If the installation finished successfully for the first time:
+        1. create the file /etc/cloudify/.installed/<service_name>.
+        2. Update the file /etc/cloudify/.installed/components.yaml with the
+           dictionary {<main_service_name>: components_installed}.
+        3. Update the file /etc/cloudify/.installed/packages.yaml with the
+           dictionary {<main_service_name>: packages}.
     """
     if not _are_components_installed():
         mkdir(INITIAL_INSTALL_DIR)
-        for service_name in get_installed_services_names():
+        for service_name in config[SERVICES_TO_INSTALL]:
```

Comment:
I think this change was made recently to avoid needing to mess about with the config on docker installs wasn't it @tehasdf ?

Reply:
Yep, I changed it to include all the installed services and not just the "main" ones (i.e. "manager_service", "database_service", "queue_service") according to the way @tehasdf uses them.

Reply:
Right, but changing it back to using the config will cause that issue again won't it?

Reply:
To be clear, the issue I'm referring to is where a bunch of components have been --only-installed, and then a config.yaml is provided for the configure stage. If that config doesn't have service_to_install provided then it'll be treated as an AIO and try to configure everything, which will include starting the entropy_service which likely won't be installed on docker containers.

The get_installed_service_names() I thought fixed that by using the list of installed services if there was one, but otherwise falling back to the config?

Reply:
If I understand the described issue correctly, then the confusion arises from the names of the different functions. @tehasdf wrote this function https://github.com/cloudify-cosmo/cloudify-manager-install/blob/d1e0d8bd8ea6ce037ddd7fc4390b40ca405977ab/cfy_manager/config.py#L61-L67 and I deleted the function that I previously wrote named `get_installed_service_names()` https://github.com/cloudify-cosmo/cloudify-manager-install/pull/940/files#diff-978e942d5c92a65a424a164f1011b7dfL263-L267.

@tehasdf function solves the issue described, by writing to the `services_to_install` section in the config.yaml the services that were installed previously, in case this section is empty. Therefore, when the `configure` command will run, the function `_are_components_installed` will return true, as all the services listed in the config.yaml (by @tehasdf function) have a corresponding file.

Reply:
@YoniItzhak are we sure we that the components.yaml file will contains the correct data ? I mean that `installed_components_list` is a list of all components from all services but for each service we are associated all components ? like on this function for each service name we do the following:  `{service_name: installed_components_list}` should each service only has its own components ?

Reply:
@mabuaisha, Yes, each service should have only its relevant components. This way when we remove a specific service from a 3 nodes cluster, we can keep the components related to the other services.